domain/src/main/java/com/thoughtworks/go/config/materials/git/GitMaterial.java

```diff
@@ -443,13 +314,136 @@ public GitMaterial withShallowClone(boolean value) {
         return gitMaterial;
     }
 
-    public String branchWithDefault() {
-        return isBlank(branch) ? GitMaterialConfig.DEFAULT_BRANCH : branch;
+    public String effectiveLocalBranch() {
+        return localBranch(isBlank(refSpecOrBranch) ? GitMaterialConfig.DEFAULT_BRANCH : refSpecOrBranch);
+    }
+
+    @Override
+    protected void appendCriteria(Map<String, Object> parameters) {
+        parameters.put(ScmMaterialConfig.URL, url.originalArgument());
+        parameters.put("branch", refSpecOrBranch);
+    }
+
+    @Override
+    protected void appendAttributes(Map<String, Object> parameters) {
+        parameters.put("url", url);
+        parameters.put("branch", refSpecOrBranch);
+        parameters.put("shallowClone", shallowClone);
+    }
+
+    @Override
+    protected String getLocation() {
+        return url.forDisplay();
     }
 
     @Override
     protected void setGoMaterialVariables(EnvironmentVariableContext environmentVariableContext) {
         super.setGoMaterialVariables(environmentVariableContext);
-        setVariableWithName(environmentVariableContext, branchWithDefault(), GO_MATERIAL_BRANCH);
+        setVariableWithName(environmentVariableContext, effectiveLocalBranch(), GO_MATERIAL_BRANCH);
+    }
+
+    private GitCommand getGit(File workingdir, SubprocessExecutionContext executionContext) {
+        InMemoryStreamConsumer output = inMemoryConsumer();
+        try {
+            return git(output, workingdir, DEFAULT_SHALLOW_CLONE_DEPTH, executionContext);
+        } catch (Exception e) {
+            throw bomb(e.getMessage() + " " + output.getStdError(), e);
+        }
+    }
+
+    private GitCommand git(ConsoleOutputStreamConsumer outputStreamConsumer, final File workingFolder, int preferredCloneDepth, SubprocessExecutionContext executionContext) throws Exception {
+        if (isSubmoduleFolder()) {
+            return new GitCommand(getFingerprint(), new File(workingFolder.getPath()), GitMaterialConfig.DEFAULT_BRANCH, true, secrets());
+        }
+        System.out.printf("**** [%s] WorkingDir for %s: %s\n", executionContext.isServer() ? "server" : "agent", urlForCommandLine(), workingFolder.getPath());
```

Comment:
@marques-work - is this `System.out.printf` intentional?

Reply:
@GaneshSPatil oops! no, it is not.